### DISCUSSION TITLE
about xr-environment

### DISCUSSION THREAD

**Discussion by ghost** [google/model-viewer#3267] [Q&A]
hi all

i have a question

i am using this code  `xr-environment` and also use `environment-image`

but my  object is too dark

how can i change my code?

```
const modelViewerVariants = document.querySelector("model-viewer#xr_box");
const select = document.querySelector('#btnBox');

modelViewerVariants.addEventListener('load', () => {
  const names = modelViewerVariants.availableVariants;
  for (const name of names) {
    const button = document.createElement('BUTTON');
    button.value = name;
    button.textContent = name;
    button.id = "btn"
    select.appendChild(button);
  }
});
console.log(modelViewerVariants)

select.addEventListener('click', (event) => {
  modelViewerVariants.variantName = event.target.value === 'default' ? null : event.target.value;
  modelViewerVariants.xrLight.environment.exposure = 3;
});
```

i dont know this part

`  modelViewerVariants.xrLight.environment.exposure = 3;`

how to change this code?

**Comment by elalish** [comment]
Honestly, `xr-environment` doesn't work very well right now; ARCore needs some work there. I'd recommend you stick with your environment image instead. To change exposure, use `modelViewerVariants.exposure = 3`, or set the exposure attribute.

**Reply by ghost** [reply]
thank you !! i have a question again.

Where can I find instructions for that code?

i will try more another code

**Reply by elalish** [reply]
https://modelviewer.dev/docs/index.html#entrydocs-lightingandenv-attributes-exposure

**Reply by ghost** [reply]
umm i saw your this "Fix xr-environment #2847"

I saw the article you edited last time. Reading that article,

I thought I could edit xrLight or change the value, isn't it?

If that's possible, I'd like to know what code I can use.

**Reply by ghost** [reply]
I think it may help to solve the problem. After we touch the UI button on the web view, the materials of AR are exactly same as the web view. I guess, when we load the materials throw `availableVariants` code, it reflects our environment image well. For your understanding, I attached the captures. The first one is when we did not touch the UI buttons, the second one is after touching the UI. Hope you give us an advice of that. Thank you for your time. 

second touched UI button

**Reply by elalish** [reply]
Well, if you change the lighting, you'd expect the colors of the model to change, just like in real life. Also, editing xrLight is outside of our public API, so you're on your own if you want to hack at that level.

**Reply by ghost** [reply]
thank you so much! i will try that
